src/bacnet/basic/object/csv.c

```diff
@@ -334,11 +403,15 @@ static char *CharacterString_Value_Description(uint32_t object_instance)
     char *pName = NULL; /* return value */
 
     index = CharacterString_Value_Instance_To_Index(object_instance);
-    if (index < MAX_CHARACTERSTRING_VALUES) {
-        pName = Object_Description[index];
+    if (index < CSV_Max_Index) {
+        pName = CSV_Descr[index].Description;
+        if(pName != NULL){
```

Comment:
I daresay you could return CSV_Descr[index].Description; and get a rid of pName entirely :)